npm/webpack-preprocessor/lib/utils.ts

```diff
@@ -0,0 +1,34 @@
+import * as os from 'os'
+import path from 'path'
+import md5 from 'md5'
+import Bluebird from 'bluebird'
+
+function createDeferred<T> () {
+  let resolve: (thenableOrResult?: T | PromiseLike<T> | undefined) => void
+  let reject: any
+  const promise = new Bluebird<T>(function (_resolve, _reject) {
+    resolve = _resolve
+    reject = _reject
+  })
+
+  return {
+    //@ts-ignore
+    resolve,
+    reject,
+    promise,
+  }
+}
+
+function hash (contents: string) {
+  return md5(contents)
+}
+
+function tmpdir () {
+  return path.join(os.tmpdir(), 'cypress', 'webpack-preprocessor')
```

Comment:
With this being a tempDir, what type of file permission testing / issue could we hit? any risk webpack will fail here?

Reply:
We use `os.tmpdir()` for [installing the Cypress binary](https://github.com/cypress-io/cypress/blob/7f2522f85769b9b72a159db6cf39807a036fba83/cli/lib/tasks/install.js#L345-L347), so if it's not writeable, the user wouldn't be able to get this far.